engine/src/test/scala/cromwell/CromwellTestkitSpec.scala

```diff
@@ -412,81 +411,47 @@ abstract class CromwellTestkitSpec extends TestKit(new CromwellTestkitSpec.TestW
+                          eventFilter: EventFilter,
+                          expectedStdoutContent: String,
+                          expectedStderrContent: String,
+                          runtime: String = "",
+                          workflowOptions: String = "{}",
+                          allowOtherOutputs: Boolean = true,
+                          terminalState: WorkflowState = WorkflowSucceeded,
+                          config: Config = DefaultConfig,
+                          workflowManagerActor: Option[TestActorRef[WorkflowManagerActor]] = None)
+                         (implicit ec: ExecutionContext): WorkflowId = {
+    val workflowManager = workflowManagerActor.getOrElse(buildWorkflowManagerActor(config))
+    val sources = sampleWdl.asWorkflowSources(runtime, workflowOptions)
+    val maxRetries = 3
+    def isFatal(e: Throwable) = e match {
+      case _: LogNotFoundException => false
+      case _ => true
     }
-  }
-
-  def runWdlWithWorkflowManagerActor(wma: TestActorRef[WorkflowManagerActor],
-                                     sources: WorkflowSourceFiles,
-                                     eventFilter: EventFilter,
-                                     stdout: Map[FullyQualifiedName, Seq[String]],
-                                     stderr: Map[FullyQualifiedName, Seq[String]],
-                                     expectedOutputs: Map[FullyQualifiedName, WdlValue] = Map.empty,
-                                     terminalState: WorkflowState = WorkflowSucceeded,
-                                     assertStdoutStderr: Boolean = false)(implicit ec: ExecutionContext) = {
     eventFilter.intercept {
       within(timeoutDuration) {
-        val workflowId = wma.submit(sources)
-        verifyWorkflowState(wma, workflowId, terminalState)
+        val workflowId = workflowManager.submit(sources)
+        verifyWorkflowState(workflowManager, workflowId, terminalState)
 
-        if (assertStdoutStderr) {
-          //val standardStreams = wma.workflowStdoutStderr(workflowId)
-          stdout foreach { ???
-//            case (fqn, out) if standardStreams.contains(fqn) =>
-//              out shouldEqual (standardStreams(fqn) map { s => File(s.stdout.value).contentAsString })
-          }
-          stderr foreach { ???
-//            case (fqn, err) if standardStreams.contains(fqn) =>
-//              err shouldEqual (standardStreams(fqn) map { s => File(s.stderr.value).contentAsString })
-          }
+        def checkLogs(): Future[WorkflowId] = Future {
+          import better.files._
+          val (stdout, stderr) = getFirstCallLogs(workflowId)
+          Paths.get(stdout).contentAsString shouldBe expectedStdoutContent
+          Paths.get(stderr).contentAsString shouldBe expectedStderrContent
+          workflowId
         }
+
+        // Retry because we have no guarantee that all the metadata is there when the workflow finishes...
+        Await.result(Retry.withRetry(
```

Comment:
@geoffjentry hopefully extractable pattern for non-cheesy eventual consistency handling, though `Retry` would need to be ~~copy/pasted~~ refactored to a common dependency